e2e/cp_test.go

```diff
@@ -4596,3 +4591,183 @@ func TestCopyS3ObjectsWithIncludeExcludeFilter2(t *testing.T) {
+		2: equals("cp %v/file1.html %sfile1.html", workdir.Path(), dst),
+		3: equals("cp %v/file2.text %sfile2.text", workdir.Path(), dst),
+		4: equals("cp %v/file3.mp4 %sfile3.mp4", workdir.Path(), dst),
+		5: equals("cp %v/file4.py %sfile4.py", workdir.Path(), dst),
+		6: equals("cp %v/readme.md %sreadme.md", workdir.Path(), dst),
+	}, sortInput(true))
+
+	// assert s3
+	assert.Assert(t, ensureS3Object(s3client, bucket, "file1.html", "<html><body><h1>file1</h1></body></html>", ensureContentType("video/avi")))
+	assert.Assert(t, ensureS3Object(s3client, bucket, "file2.text", "this is a test file 2", ensureContentType("video/avi")))
+	assert.Assert(t, ensureS3Object(s3client, bucket, "file3.mp4", "this is a test file 3", ensureContentType("video/avi")))
+	assert.Assert(t, ensureS3Object(s3client, bucket, "file4.py", "import tensorflow as tf", ensureContentType("video/avi")))
+	assert.Assert(t, ensureS3Object(s3client, bucket, "readme.md", "this is a readme file", ensureContentType("video/avi")))
+	assert.Assert(t, ensureS3Object(s3client, bucket, "a/another_test_file.txt", "yet another txt file. yatf.", ensureContentType("video/avi")))
+	assert.Assert(t, ensureS3Object(s3client, bucket, "b/filename-with-hypen.gz", "file has hypen in its name", ensureContentType("video/avi")))
+}
+
+// cp --content-type "video/avi" s3://srcbucket/object s3://dstbucket/
+func TestCopySingleS3ObjectToAnotherBucketWithContentType(t *testing.T) {
+	t.Parallel()
+
+	srcbucket := s3BucketFromTestNameWithPrefix(t, "src")
+	dstbucket := s3BucketFromTestNameWithPrefix(t, "dst")
+
+	s3client, s5cmd := setup(t)
+
+	createBucket(t, s3client, srcbucket)
+	createBucket(t, s3client, dstbucket)
+
+	const (
+		filename = "testfile.txt"
+		content  = "this is a test file"
+	)
+
+	putFile(t, s3client, srcbucket, filename, content)
+
+	src := fmt.Sprintf("s3://%v/%v", srcbucket, filename)
+	dst := fmt.Sprintf("s3://%v/", dstbucket)
+
+	cmd := s5cmd("cp", "--content-type", "video/avi", src, dst)
+	result := icmd.RunCmd(cmd)
+
+	result.Assert(t, icmd.Success)
+
+	// assert s3
+	assert.Assert(t, ensureS3Object(s3client, srcbucket, filename, content))
+	assert.Assert(t, ensureS3Object(s3client, dstbucket, filename, content, ensureContentType("video/avi")))
+}
+
+// cp --content-type "video/avi" s3://srcbucket/* s3://dstbucket/
+func TestCopyMultipleS3ObjectsToAnotherBucketWithContentType(t *testing.T) {
+	t.Parallel()
+
+	srcbucket := s3BucketFromTestNameWithPrefix(t, "src")
+	dstbucket := s3BucketFromTestNameWithPrefix(t, "dst")
+
+	s3client, s5cmd := setup(t)
+
+	createBucket(t, s3client, srcbucket)
+	createBucket(t, s3client, dstbucket)
+
+	fileAndContent := map[string]string{
+		"file1.txt":                "this is a test file 1",
+		"file2.html":               "<html><body><h1>file2</h1></body></html>",
+		"file3.mp4":                "this is a test file 3",
+		"file4.py":                 "import tensorflow as tf",
```

Comment:
Four files are not needed to assert that content type is set to the desired value. One file is enough. Adding one for flat, one for nested case is sufficient. It becomes harder to main the tests after some point when other stuff keeps testing the same thing over and over, adding tests that cover the possible cases and leaving at that point is more sustainable :)

Reply:
done.